assets/js/plugin-check-admin.js

```diff
@@ -207,4 +222,108 @@
 
 		return data;
 	}
+
+	/**
+	 * Renders results for each check on the page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} results The results object.
+	 */
+	function renderResults( results ) {
+		const { errors, warnings } = results;
+
+		// Render errors and warnings for files.
+		for ( const file in errors ) {
+			if ( warnings[ file ] ) {
+				renderFileResults( file, errors[ file ], warnings[ file ] );
+				delete warnings[ file ];
+			} else {
+				renderFileResults( file, errors[ file ], [] );
+			}
+		}
+
+		// Render remaining files with only warnings.
+		for ( const file in warnings ) {
+			renderFileResults( file, [], warnings[ file ] );
+		}
+
+		resultsContainer.innerHTML +=
+			'<p>' + wp.i18n.__( 'Checks complete', 'plugin-check' ) + '</p>';
```

Comment:
Providing translations to JS is a bit trickier, and since this is the only translated string we have in JS, I would prefer if we handled that in PHP and made it available to JS.

While that may seem a bit overkill, why don't we put this `<p>` tag with the string in another `script[type="text/template"]` and then render it like that? That would then fully decouple the UI from the JS logic.